source/crud/crud.rst

```diff
@@ -1253,6 +1253,14 @@ Below are defined the exceptions that should be thrown from the various write me
     unprocessedRequests: Optional<Iterable<WriteModel>>;
 
     /**
+     * The intermediary write result for any operations that succeeded before
+     * the bulk write was interrupted.
+     *
+     * NOT REQUIRED: Drivers may choose to not provide this property.
+     */
+    writeResult: Optional<BulkWriteResult|InsertManyResult>;
```

Comment:
@jyemin, @rstam: Please let me know if this needs more wording for statically typed languages. IIRC, @craiggwilson told me on Friday that C# already provides access to an intermediary write result on its BulkWriteException ([here](https://github.com/mongodb/mongo-csharp-driver/blob/985e8e618868de9dd10b26b480b48eed5e7ff5b1/src/MongoDB.Driver/MongoBulkWriteException.cs#L176-L182)?).

Reply:
A property can only have one type anyway, so let's just call it a BulkWriteResult and leave InsertManyResult out of it.

Reply:
Will do. I thought it best to be permissive here because earlier in the spec both `bulkWrite()` and `insertMany()` are documented as throwing a BulkWriteException despite the fact that they return a BulkWriteResult and InsertManyResult, respectively.